packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx

```diff
@@ -255,6 +265,9 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
   }
 
   private _onNavButtonLinkClicked(link: INavLink, ev: React.MouseEvent<HTMLElement>): void {
+    // In this case, the href would be "#" and needs a preventDefault to prevent it from bouncing around
```

Comment:
Not sure this is needed here, since this click handler should only be used if the link is a button (and therefore href wouldn't be respected).

Reply:
so, in our props, we allow people to configure a forceAnchor: true WITH a onclick (such is the case in the OP's codepen I think). We must handle this as well.